draft-ietf-quic-v2.md

```diff
@@ -190,6 +190,18 @@ The client MUST NOT send 0-RTT packets using the negotiated version, even after
 processing a packet of that version from the server. Servers can apply original
 version 0-RTT packets to a connection without additional considerations.
 
+# TLS Resumption
+
+TLS session tickets are specific to the QUIC version of the connection that
+provided them. Clients MUST NOT use a session ticket from a QUICv1 connection
+to initiate a QUICv2 connection, or vice versa.
+
+Servers MAY validate the originating version of any session ticket and reject
```

Comment:
```suggestion
Servers SHOULD validate the originating version of any session ticket and reject
```

Reply:
Here's the hard part...

```suggestion
Servers MUST validate the originating version of any session ticket and not resume from
```

The easy bit: rejection isn't really the right term to use.  It's not resuming.

Second, this is where the rubber hits the road.  The general strategy in TLS is to allow clients to attempt ticket use anywhere, on the understand that tickets won't be accepted by anything but the correct server.

Insisting on clients doing the right thing is fairly easy, so I'm OK with the "MUST" above, but if you are going to do this right, the "MUST" really belongs on the server side.

As for concerns @huitema raised about being able to implement this, I don't see how you can avoid having the machinery necessary to implement some sort of application/transport state if you are supporting arbitrary application protocols - or 0-RTT.  As soon as you do 0-RTT, you need some way to reclaim state.  And many application protocols need some way to carry state across resumptions.  HTTP does.  You might be able to avoid it with very careful rotation of STEKs and other such tricks, but carrying state really is the easiest way.

Reply:
MT convinced me, let's make this a MUST.